Change scheduler to be used only for state observing

OhKanghoon opened the pull request:
- Change scheduler to be used only for state observing
- Fixes #149

OhKanghoon committed:
feat: change scheduler to be used only for state observing

OhKanghoon committed:
docs: change a documentation for scheduling

tokijh reviewed (commented):
How about dividing it into `stateScheduler` and `scheduler`?

tokijh reviewed (approved):
Dividing `scheduler` and `stateScheduler` will be discussed separately.
In private, we discussed setting MainScheduler as the default.

OhKanghoon committed:
feat: update default scheduler

OhKanghoon commented:
I agree that the mutate, reduce method can be a problem when called from the concurrent thread
However, I prefer that Reactor does not involve much of scheduling like [TCA](https://github.com/pointfreeco/swift-composable-architecture).
This discussion will continue. If you have any ideas, please leave a comment.

In the case of scheduler, we modified it to be used only for the state, just like the original design.
I changed the scheduler default value to `MainScheduler.instance` because it is commonly used by main threa.

OhKanghoon committed:
docs: update markup
